Replace the logging.basicConfig by a proper configuration

evandrocoan opened the pull request:
basicConfig: Do basic configuration for the logging system.
This function does nothing if the root logger already has handlers
configured. It is a convenience method intended for use by simple
scripts to do one-shot configuration of the logging package.

evandrocoan committed:
Replace the logging.basicConfig by a proper configuration

basicConfig: Do basic configuration for the logging system.
This function does nothing if the root logger already has handlers
configured. It is a convenience method intended for use by simple
scripts to do one-shot configuration of the logging package.

gatesn commented on pyls/__main__.py:
```diff
@@ -95,3 +80,37 @@ def _binary_stdio():
+        log_file = args.log_file
+        formatter = logging.Formatter(LOG_FORMAT, style="%")
+        if log_file:
+            log_handler = logging.handlers.RotatingFileHandler(log_file, mode='a',
```
Can you stick to repo parenthesis style? 
```
logging.handlers.RotatingFileHandler(
    log_file, mode='a', maxBytes=50*1024*1024, 
    backupCount=10, encoding=None, delay=0
)
```

evandrocoan replied:
Oops.

gatesn commented on pyls/__main__.py:
```diff
@@ -95,3 +80,37 @@ def _binary_stdio():
+    # ERROR = 40
+    # WARNING = 30
+    # INFO = 20
+    # DEBUG = 10
```
Why are these comments here? They don't seem to be relevant to the code

evandrocoan replied:
Informative about the debug levels. I put it there because I was confused about them when I saw you can only pass 3 levels, but I know the `logging` module has 6. So, looking at these values I can clearly see that we you set the WARNING level, you can also see the ERROR and other above it.

gatesn replied:
I think that's part of the logging API and doesn't need re-documenting here. Most logging frameworks would log at the given level or "above"

gatesn commented on pyls/__main__.py:
```diff
@@ -95,3 +80,37 @@ def _binary_stdio():
 
     return stdin, stdout
+
+def configure_logger(args):
```
Can you prefix function with underscore

evandrocoan replied:
Sure.

gatesn commented:
Some linting errors too:

```
************* Module pyls.__main__
C: 96, 0: Wrong continued indentation (add 43 spaces).
                    maxBytes=50*1024*1024, backupCount=10, encoding=None, delay=0)
                    ^                                          | (bad-continuation)
C:116, 0: Trailing newlines (trailing-newlines)
E: 93,20: Unexpected keyword argument 'style' in constructor call (unexpected-keyword-arg)
I: 75, 0: Useless suppression of 'no-member' (useless-suppression)
I: 75, 0: Useless suppression of 'import-error' (useless-suppression)
```

evandrocoan commented:
I should have run `tox` before. I am running it now.

evandrocoan committed:
Renamed configure_logger(args) and fixed multiline function call.

evandrocoan committed:
Fix linter warnings C:118 and I: 68

************* Module pyls.__main__
C:118, 0: Trailing newlines (trailing-newlines)
I: 68, 0: Useless suppression of 'no-member' (useless-suppression)

-----------------------------------
Your code has been rated at 9.99/10

evandrocoan committed:
Removed python 3 only parameter for logging.Formatter(style="%")

evandrocoan committed:
pyls\__main__.py:84:1: E302 expected 2 blank lines, found 1

evandrocoan commented:
@gatesn, this linter is insane:

1. First it complained about missing 2 trailing new lines. 
1. Now it complains about a existent trailing new line.

gatesn committed:
Update __main__.py

gatesn committed:
Update __main__.py
